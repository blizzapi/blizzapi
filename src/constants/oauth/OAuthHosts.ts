/**
 * List of all available OAuth token URIs
 * via https://develop.battle.net/documentation/guides/using-oauth
 */

import {
  RegionIdProperties,
  Uri,
} from '../../types';

const oAuthHosts: RegionIdProperties<Uri> = {
  1: 'https://us.battle.net',
  2: 'https://eu.battle.net',
  3: 'https://apac.battle.net',
  5: 'https://www.battlenet.com.cn',
};

export default Object.freeze(oAuthHosts);
