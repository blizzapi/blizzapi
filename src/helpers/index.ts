export * from "./bnet";
export * from "./fetch";
export * from "./oauth";
export * from "./validators";
