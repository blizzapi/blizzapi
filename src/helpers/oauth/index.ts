export { getAccessTokenObject } from "./getAccessTokenObject";
export { validateAccessToken } from "./validateAccessToken";
