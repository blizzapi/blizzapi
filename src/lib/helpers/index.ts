export default {
  query: (endpoint: string) => {
    console.log(`query: ${endpoint}`);
  }
}