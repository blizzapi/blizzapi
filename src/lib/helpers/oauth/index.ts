export { default as fetchAccessToken } from './fetchAccessToken';
export { default as getAccessToken } from './getAccessToken';
export { default as validateAccessToken } from './validateAccessToken';
