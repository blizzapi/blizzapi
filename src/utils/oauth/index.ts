export * from "./authorizeUris";
export * from "./checkTokenUris";
export * from "./tokenUris";
