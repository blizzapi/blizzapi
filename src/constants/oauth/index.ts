import { oAuthHosts } from "./OAuthHosts";
import { RegionIdProperties, RegionId, Uri } from "../../types";

const getRegionPropertyArray = (endpoint: string): RegionIdProperties<Uri> =>
  Object.freeze({
    [RegionId.us]: `${oAuthHosts[RegionId.us]}${endpoint}`,
    [RegionId.eu]: `${oAuthHosts[RegionId.eu]}${endpoint}`,
    [RegionId.kr]: `${oAuthHosts[RegionId.kr]}${endpoint}`,
    [RegionId.cn]: `${oAuthHosts[RegionId.cn]}${endpoint}`,
  });

const authorizeEndpoint = "/oauth/authorize";
const tokenEndpoint = "/oauth/token";
const checkTokenEndpoint = "/oauth/check_token?token=";

export const authorizeUris = getRegionPropertyArray(authorizeEndpoint);
export const tokenUris = getRegionPropertyArray(tokenEndpoint);
export const checkTokenUris = getRegionPropertyArray(checkTokenEndpoint);
