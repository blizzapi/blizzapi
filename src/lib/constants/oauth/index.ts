import { RegionIdProperties } from '../../../../@interfaces';
import { Uri } from '../../../../@types';
import OAuthHosts from './OAuthHosts';

const getRegionPropertyArray = (endpoint: string): RegionIdProperties<Uri> => ({
  1: `${OAuthHosts[1]}${endpoint}`,
  2: `${OAuthHosts[2]}${endpoint}`,
  3: `${OAuthHosts[3]}${endpoint}`,
  5: `${OAuthHosts[5]}${endpoint}`,
});

const authorizeEndpoint = '/oauth/authorize';
const tokenEndpoint = '/oauth/token';
const checkTokenEndpoint = '/oauth/check_token?token=';

const authorizeUris = getRegionPropertyArray(authorizeEndpoint);
const tokenUris = getRegionPropertyArray(tokenEndpoint);
const checkTokenUris = getRegionPropertyArray(checkTokenEndpoint);

export default {
  authorizeUris: Object.freeze(authorizeUris),
  tokenUris: Object.freeze(tokenUris),
  checkTokenUris: Object.freeze(checkTokenUris),
}
