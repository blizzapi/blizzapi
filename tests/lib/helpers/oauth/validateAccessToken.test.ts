import { validateAccessToken } from '../../../../src/lib/helpers/oauth';

// import regionNamesJson from '../../../__testData__/regionNames.json';
// import wrongRegionNamesJson from '../../../__testData__/wrongRegionNames.json';
// import regionIdsJson from '../../../__testData__/regionIds.json';
// import wrongRegionIdsJson from '../../../__testData__/wrongRegionIds.json';

const fetch = require('../../../../src/lib/helpers/fetch');

// tslint:disable-next-line: no-object-mutation
fetch.fetchFromUri = jest.fn().mockImplementation((options: { uri: string }) => {
  if (options.uri.includes("incorrectAccessToken")) {
    return {
      error: 'invalid_token',
    }
  }
  if (options.uri.includes("accessTokenWithApiError")) {
    throw new Error('invalid token');
  }
  return {
    data: 'valid token',
  }
});

describe('validateAccessToken()', () => {
  const region = 'us';
  const accessToken = 'correctAccessToken';
  const incorrectAccessToken = 'incorrectAccessToken';
  const accessTokenWithApiError = 'accessTokenWithApiError';

  test('should be defined', () => {
    expect(validateAccessToken).toBeDefined();
  });

  test('should be function', () => {
    expect(validateAccessToken).toBeTruthy();
    expect(typeof validateAccessToken).toBe('function');
  });

  test('returns true for correct token', async () => {
    const response = await validateAccessToken(region, accessToken);
    expect(response).toMatchSnapshot();
  });

  test('returns false for incorrect token', async () => {
    const response = await validateAccessToken(region, incorrectAccessToken);
    expect(response).toMatchSnapshot();
  });

  test('returns false in case of API error', async () => {
    const response = await validateAccessToken(region, accessTokenWithApiError);
    expect(response).toMatchSnapshot();
  });
});
