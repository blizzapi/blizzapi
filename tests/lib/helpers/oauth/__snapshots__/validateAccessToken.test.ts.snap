// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`validateAccessToken() returns false for invalid access token 1`] = `false`;

exports[`validateAccessToken() returns false in case of API error 1`] = `false`;

exports[`validateAccessToken() returns true for valid access token 1`] = `true`;
